src/Compilers/Test/Core/Assert/AssertEx.cs

```diff
@@ -806,7 +806,7 @@ public static void Multiple(bool includeStackTrace, params Action[] assertions)
                 return;
 
             var stringBuilder = new StringBuilder($"{exceptions.Count} out of {assertions.Length} assertions failed.");
-            foreach (var (index, ex) in exceptions)
+            foreach (var (index, ex) in exceptions.OrderBy(e => e.Index))
```

Comment:
If we need to do this (why?) there's no need to run assertions in the reverse order above.